server/src/main/java/org/schemarepo/server/JsonRenderer.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.schemarepo.server;
+
+import java.util.Properties;
+
+import javax.ws.rs.core.MediaType;
+
+import org.schemarepo.RepositoryUtil;
+import org.schemarepo.SchemaEntry;
+import org.schemarepo.Subject;
+import org.schemarepo.json.JsonUtil;
+
+/**
+ * Renders as JSON. The actual work is delegated to utility class {@link org.schemarepo.RepositoryUtil}
+ */
+public class JsonRenderer implements Renderer {
+
+  private final JsonUtil jsonUtil;
+
+  public JsonRenderer(JsonUtil jsonUtil) {
+    if (jsonUtil == null) {
+      throw new IllegalArgumentException("jsonUtil required");
+    }
+    this.jsonUtil = jsonUtil;
+  }
+
+  @Override
+  public String getMediaType() {
+    return MediaType.APPLICATION_JSON;
+  }
+
+  @Override
+  public String renderSubjects(Iterable<Subject> subjects) {
+    return jsonUtil.subjectsToJson(subjects);
+  }
+
+  @Override
+  public String renderSchemas(Iterable<SchemaEntry> schemaEntries) {
+    return jsonUtil.schemasToJson(schemaEntries);
+  }
+
+  @Override
+  public String renderSchemaEntry(SchemaEntry schemaEntry, boolean requestForLatest) {
+    return requestForLatest ? schemaEntry.toString() : schemaEntry.getSchema();
+  }
+
+  @Override
+  public String renderProperties(Properties props, String comment) {
+    // TODO: when the need arises, this may yield an actual JSON
```

Comment:
Thanks, you bring up a very good point @eric239.

I did a bit of digging and here is what I found out:

In master:

* ```415 Unsupported Media Type``` responses are only used for write operations where the ```Content-Type``` header is inappropriate. I think this generally makes sense (though we may want to be a bit more lax in certain cases, as requested in Issue #3, but let's ignore that for the purpose of this discussion).
* Read operations apparently "never fail", no matter what the ```Accept``` header is. This second point surprised me because I vaguely recalled getting some errors in the past. It does mean your previous comment is right, however.

Before the current master:

It turns out that I changed the behavior in 4932f5494d29d461cab746b4667b728f06cccb5e (as part of the work on Issue #17 to support multi-line schemas using JSON responses). Before then, the server would return ```406 Not Acceptable``` when an unsupported ```Accept``` header was requested. My memory failed me in that it wasn't previously returning ```415```, but I think this old behavior (```406```) was more proper than the current one (not honoring the request's ```Accept``` header and making assumptions). I should have been more careful when working on the multi-line issue.

I believe the old behavior happened because the ```RESTRepository``` was prefixed by the ```@Produces(MediaType.TEXT_PLAIN)``` annotation, which I then removed as part of 4932f5494d29d461cab746b4667b728f06cccb5e. By the way, when running 4932f5494d29d461cab746b4667b728f06cccb5e, doing a request with ```Accept: text/html``` also returns a ```406``` error code, so I'm not sure what's the deal with what you were saying about the Jersey client always requesting text/html? (I haven't looked into verifying that...)

I would like it if we could bring back the old behavior, though I'm not sure if it should be done by resuming to use ```@Produces``` annotations (which are completely eliminated in the current master), or if we should somehow manually throw 406 errors. At first glance, I haven't seen a Jersey exception which represents 406, so I don't know if there is any way of programmatically triggering those, or if the ```@Produces``` annotations are the only way...

This is a bit of a mess, and my fault. I'm willing to put some time in fixing it (unless you want to), but as of now I'm not sure what the best solution would be. I would need to read up and tinker some more...

Thoughts?